models/media_ad_views/scratch/snowplow_media_player_media_ad_views_this_run.sql

```diff
@@ -0,0 +1,68 @@
+{{
+  config(
+    materialized='table',
+    tags=["this_run"],
+    sort='last_event',
+    dist='media_ad_id',
+    partition_by = snowplow_utils.get_value_by_target_type(bigquery_val={
+      "field": "viewed_at",
+      "data_type": "timestamp"
+    }, databricks_val='viewed_at_date'),
+    cluster_by=snowplow_utils.get_value_by_target_type(bigquery_val=["media_ad_id"]),
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with
+
+events_this_run as (
+
+    select * from {{ ref('snowplow_media_player_base_events_this_run') }}
+    where ad_id is not null and media_id is not null
+
+)
+
+, prep as (
+
+  select
+    {{ dbt_utils.generate_surrogate_key(['ev.platform', 'ev.media_id', 'ev.ad_id']) }} as media_ad_id,
```

Comment:
Curious what the reason is for including the plaform in this key is? Why should it matter what platform a media and ad was viewed on to not group them?

Reply:
I wasn't sure about this decision, went for including the platform as it's easier for users to aggregate the platforms if they are not interested in them rather than going the other way. So was just being careful in case people want to measure ad performance differently on web and mobile apps, but happy to change it.